### DISCUSSION TITLE
nzyme-tap_2.0.0-alpha.5_rpios11_arm64

### DISCUSSION THREAD

**Discussion by collote69** [nzymedefense/nzyme#968] [Questions] [answered]
Hello and installed nzyme-tap_2.0.0-alpha.5_rpios11_arm64.deb On raspberry pi 4, I follow all the configuration of the page, for nzyme-tap.conf 2 AWUS036NEH wifi cards and when I sudo systemctl start nzyme-tap, it does this.

● nzyme-tap.service - Nzyme-Tap
     Loaded: loaded (/lib/systemd/system/nzyme-tap.service; enabled; vendor preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Sun 2023-11-26 11:55:22 CET; 7s ago
       Docs: https://github.com/lennartkoopmann/nzyme
    Process: 4346 ExecStart=/usr/bin/nzyme-tap --configuration-file /etc/nzyme/nzyme-tap.conf --log-level info (code=exited, status=71)
   Main PID: 4346 (code=exited, status=71)
        CPU: 139ms

**when doing journalctl -xe**

Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [INFO][2023-11-26 11:42:30][ethernet::capture] Available network device: wlp1s0u1u4 (no description): [192.168.0.109, fe>
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [INFO][2023-11-26 11:42:30][ethernet::capture] Available network device: wlp1s0u1u3 (no description): [192.168.0.183, fe>
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [INFO][2023-11-26 11:42:30][ethernet::capture] Available network device: any (Pseudo-device that captures on all interfa>
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [INFO][2023-11-26 11:42:30][ethernet::capture] Available network device: lo (no description): [127.0.0.1, ::1]
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [INFO][2023-11-26 11:42:30][ethernet::capture] Available network device: bluetooth0 (Bluetooth adapter number 0): [no in>
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [INFO][2023-11-26 11:42:30][ethernet::capture] Available network device: bluetooth-monitor (Bluetooth Linux Monitor): [n>
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [INFO][2023-11-26 11:42:30][ethernet::capture] Available network device: nflog (Linux netfilter log (NFLOG) interface): >
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [INFO][2023-11-26 11:42:30][ethernet::capture] Available network device: nfqueue (Linux netfilter queue (NFQUEUE) interf>
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [INFO][2023-11-26 11:42:30][ethernet::capture] Available network device: dbus-system (D-Bus system bus): [no interface a>
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [INFO][2023-11-26 11:42:30][ethernet::capture] Available network device: dbus-session (D-Bus session bus): [no interface>
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [INFO][2023-11-26 11:42:30][brokers::ethernet_broker] Installing ethernet broker thread <0>.
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [INFO][2023-11-26 11:42:30][brokers::dot11_broker] Installing WiFi broker thread <0>.
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [INFO][2023-11-26 11:42:30][ethernet::capture] Starting ethernet capture on [eth0].
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [INFO][2023-11-26 11:42:30][dot11::capture] Starting WiFi capture on [wlp1s0u1u2]
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [INFO][2023-11-26 11:42:30][dot11::capture] Starting WiFi capture on [wlp1s0u1u1]
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [INFO][2023-11-26 11:42:30][dot11::capture] Temporarily disabling interface [wlp1s0u1u2] ...
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [INFO][2023-11-26 11:42:30][dot11::capture] Temporarily disabling interface [wlp1s0u1u1] ...
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [ERROR][2023-11-26 11:42:30][dot11::capture] Could not disable device [wlp1s0u1u2]: Could not load device with name [wlp>
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [ERROR][2023-11-26 11:42:30][nzyme_tap] WiFi capture [wlp1s0u1u2] disconnected. Retrying in 5 seconds.
Nov 26 11:42:30 raspberrypi kernel: device eth0 entered promiscuous mode
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [ERROR][2023-11-26 11:42:30][dot11::capture] Could not disable device [wlp1s0u1u1]: Could not load device with name [wlp>
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [ERROR][2023-11-26 11:42:30][dot11::channel_hopper] Could not fetch information of device [wlp1s0u1u2]. Not assigning to>
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [ERROR][2023-11-26 11:42:30][nzyme_tap] WiFi capture [wlp1s0u1u1] disconnected. Retrying in 5 seconds.
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [ERROR][2023-11-26 11:42:30][dot11::channel_hopper] Could not fetch information of device [wlp1s0u1u1]. Not assigning to>
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [INFO][2023-11-26 11:42:30][dot11::channel_hopper] Available adapters and channels: {}
Nov 26 11:42:30 raspberrypi nzyme-tap[3464]: [ERROR][2023-11-26 11:42:30][nzyme_tap] Could not initialize ChannelHopper: WiFi adapter [wlp1s0u1u2] not found.
Nov 26 11:42:31 raspberrypi kernel: device eth0 left promiscuous mode
Nov 26 11:42:31 raspberrypi systemd[1]: nzyme-tap.service: Main process exited, code=exited, status=71/OSERR
░░ Subject: Unit process exited
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ An ExecStart= process belonging to unit nzyme-tap.service has exited.
░░ 
░░ The process' exit code is 'exited' and its exit status is 71.
Nov 26 11:42:31 raspberrypi systemd[1]: nzyme-tap.service: Failed with result 'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support

I don't know what can happen?

Thank you

**Comment by collote69** [comment, marked as answer]
Resolved

**Comment by AlexVonLoon** [comment]
Can you provide the solution? Similar problem.. Almost identical log.

**Reply by lennartkoopmann** [reply]
Hi! Not OP, but this looks like an issue with the WiFi adapter hardware. What type of adapter are you using?

**Comment by AlexVonLoon** [comment]
I'm using ALFA AWUS036ACS. I followed the instructions from crs369 post.

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by lennartkoopmann** [reply]
Can you post your log file or errors you see?

**Reply by AlexVonLoon** [reply]
░░ Subject: Unit process exited
░░ Defined-By: systemd
░░ Support: http://www.ubuntu.com/support
░░
░░ An ExecStart= process belonging to unit nzyme-tap.service has exited.
░░
░░ The process' exit code is 'exited' and its exit status is 71.
May 16 13:53:02 nzyme systemd[1]: nzyme-tap.service: Failed with result 'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: http://www.ubuntu.com/support
░░
░░ The unit nzyme-tap.service has entered the 'failed' state with result 'exit-code'.
lines 1842-1881/1881 (END)
░░ A start job for unit nzyme-tap.service has finished successfully.
░░
░░ The job identifier is 44872.
May 16 13:53:02 nzyme nzyme-tap[28231]: [INFO][2025-05-16 13:53:02][logging] Initialized logging with level filter [INFO].
May 16 13:53:02 nzyme nzyme-tap[28231]: [INFO][2025-05-16 13:53:02][nzyme_tap] Starting nzyme tap version [2.0.0-alpha.17].
May 16 13:53:02 nzyme nzyme-tap[28231]: [INFO][2025-05-16 13:53:02][nzyme_tap] Parsed and loaded configuration.
May 16 13:53:02 nzyme nzyme-tap[28231]: [INFO][2025-05-16 13:53:02][wired::interfaces] Available network device: ens33 (no description): [192.168.1.120, fd64:9220:d18e:fabb:20c:29ff:fe2a:8384, fe80::20c:29ff:fe2a:8384]
May 16 13:53:02 nzyme nzyme-tap[28231]: [INFO][2025-05-16 13:53:02][wired::interfaces] Available network device: any (Pseudo-device that captures on all interfaces): [no interface addresses]
May 16 13:53:02 nzyme nzyme-tap[28231]: [INFO][2025-05-16 13:53:02][wired::interfaces] Available network device: lo (no description): [127.0.0.1, ::1]
May 16 13:53:02 nzyme nzyme-tap[28231]: [INFO][2025-05-16 13:53:02][wired::interfaces] Available network device: docker0 (no description): [172.17.0.1]
May 16 13:53:02 nzyme nzyme-tap[28231]: [INFO][2025-05-16 13:53:02][wired::interfaces] Available network device: wlx00c0cab7c753 (no description): [no interface addresses]
May 16 13:53:02 nzyme nzyme-tap[28231]: [INFO][2025-05-16 13:53:02][wired::interfaces] Available network device: bluetooth-monitor (Bluetooth Linux Monitor): [no interface addresses]
May 16 13:53:02 nzyme nzyme-tap[28231]: [INFO][2025-05-16 13:53:02][wired::interfaces] Available network device: nflog (Linux netfilter log (NFLOG) interface): [no interface addresses]
May 16 13:53:02 nzyme nzyme-tap[28231]: [INFO][2025-05-16 13:53:02][wired::interfaces] Available network device: nfqueue (Linux netfilter queue (NFQUEUE) interface): [no interface addresses]
May 16 13:53:02 nzyme nzyme-tap[28231]: [INFO][2025-05-16 13:53:02][wired::interfaces] Available network device: dbus-system (D-Bus system bus): [no interface addresses]
May 16 13:53:02 nzyme nzyme-tap[28231]: [INFO][2025-05-16 13:53:02][wired::interfaces] Available network device: dbus-session (D-Bus session bus): [no interface addresses]
May 16 13:53:02 nzyme nzyme-tap[28231]: [INFO][2025-05-16 13:53:02][state::state] Initializing state.
May 16 13:53:02 nzyme nzyme-tap[28231]: [INFO][2025-05-16 13:53:02][context::context_engine] Initializing context engine.
May 16 13:53:02 nzyme nzyme-tap[28231]: [INFO][2025-05-16 13:53:02][wired::ethernet::ethernet_broker] Installing ethernet broker thread <0>.
May 16 13:53:02 nzyme nzyme-tap[28231]: [INFO][2025-05-16 13:53:02][nzyme_tap] Skipping disabled WiFi interface [wlx00c0ca000001].
May 16 13:53:02 nzyme nzyme-tap[28231]: [INFO][2025-05-16 13:53:02][wireless::dot11::dot11_broker] Installing WiFi broker thread <0>.
May 16 13:53:02 nzyme nzyme-tap[28231]: [INFO][2025-05-16 13:53:02][wired::ethernet::capture] Starting ethernet capture on [enp6s0].
May 16 13:53:02 nzyme nzyme-tap[28231]: [ERROR][2025-05-16 13:53:02][wireless::dot11::channel_hopper] Could not fetch information of device [wlx00c0ca000000]. Not assigning to channels: Interface [wlx00c0ca000000] not found.
May 16 13:53:02 nzyme nzyme-tap[28231]: [INFO][2025-05-16 13:53:02][wireless::dot11::channel_hopper] Skipping disabled WiFi interface [wlx00c0ca000001].
May 16 13:53:02 nzyme nzyme-tap[28231]: [INFO][2025-05-16 13:53:02][wireless::dot11::channel_hopper] Available adapters and channels: {}
May 16 13:53:02 nzyme nzyme-tap[28231]: [ERROR][2025-05-16 13:53:02][nzyme_tap] Could not initialize ChannelHopper: WiFi adapter [wlx00c0ca000000] not found.
May 16 13:53:02 nzyme systemd[1]: nzyme-tap.service: Main process exited, code=exited, status=71/OSERR
░░ Subject: Unit process exited
░░ Defined-By: systemd
░░ Support: http://www.ubuntu.com/support
░░
░░ An ExecStart= process belonging to unit nzyme-tap.service has exited.
░░
░░ The process' exit code is 'exited' and its exit status is 71.
May 16 13:53:02 nzyme systemd[1]: nzyme-tap.service: Failed with result 'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: http://www.ubuntu.com/support
░░
░░ The unit nzyme-tap.service has entered the 'failed' state with result 'exit-code'.

**Reply by lennartkoopmann** [reply]
Can you also post your tap configuration file?

**Comment by AlexVonLoon** [comment]
[general]
leader_secret = "*********************************************************"
leader_uri = "https://192.168.1.120:22900/"
accept_insecure_certs = true

[ethernet_interfaces.enp6s0]
active = true
networks = [
  { cidr = "192.168.1.0/24", dns_servers = [ "192.168.1.1:53" ], injection_interface = "enp6s0" }
]

[wifi_interfaces.wlx00c0ca000000]
active = true
channel_width_hopping_mode = "full" # Learn more: https://go.nzyme.org/wifi-hopping-channel-width-mode
channels_2g = [1, 6, 11]
channels_5g = [100, 102, 104, 106, 108, 110, 112, 136, 138, 140, 149, 151, 153, 155, 157, 159, 161, 165]
channels_6g = []

[wifi_interfaces.wlx00c0ca000001]
active = false
channel_width_hopping_mode = "full" # Learn more: https://go.nzyme.org/wifi-hopping-channel-width-mode
channels_2g = [2, 3, 4, 5, 7, 8, 9, 10, 12, 13]
channels_5g = [36, 40, 44, 48, 52, 56, 60, 64, 173]
channels_6g = [5, 9, 13, 17, 21, 25, 29, 109, 113, 153, 157, 161, 165, 169, 173, 177, 181, 185, 189, 229, 233]

[bluetooth_interfaces.hci0]
active = true
bt_classic_enabled = true
bt_le_enabled = true
discovery_period_seconds = 15
dbus_method_call_timeout_seconds = 2

[performance]
ethernet_brokers = 1
wifi_brokers = 1
wifi_broker_buffer_capacity = 65535
ethernet_broker_buffer_capacity = 65535
bluetooth_devices_pipeline_size = 1024

[protocols.tcp]
pipeline_size = 16384
reassembly_buffer_size = 1048576
session_timeout_seconds = 43200

[protocols.udp]
pipeline_size = 16384

[protocols.dns]
pipeline_size = 4096
entropy_zscore_threshold = 3.0

[protocols.ssh]
pipeline_size = 1024

[protocols.socks]
pipeline_size = 1024

[protocols.arp]
pipeline_size = 1024

[protocols.dhcpv4]
pipeline_size = 1024

[protocols.uav_remote_id]
pipeline_size = 1024

[misc]
training_period_minutes = 5
context_mac_ip_retention_hours = 36
context_mac_hostname_retention_hours = 36

**Reply by lennartkoopmann** [reply]
Looks like you copied `wlx00c0ca000000` and `wlx00c0ca000001` from the example config file. Those need to be set to the actual names of your WiFi adapters.

**Reply by AlexVonLoon** [reply]
Oooooh :facepalm:

Thanks, you're a lifesaver. Up and running now.

**Reply by lennartkoopmann** [reply]
Perfect! No problem.
